tests/wpt/mozilla/tests/mozilla/mozbrowser/private_browsing.html

```diff
@@ -0,0 +1,36 @@
+<!doctype html>
+<html>
+<head>
+<meta charset="utf8" />
+<title>Private browsing</title>
+</head>
+<body>
+<iframe id="private_iframe"></iframe>
+</body>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+  async_test(function(t) {
+    var privateFrame = document.getElementById("private_iframe");
+    privateFrame.mozbrowser = "true";
+    privateFrame.mozprivatebrowsing = "true";
+
+    privateFrame.onload = t.step_func(function() {
+      var parent = privateFrame.parentNode;
+      parent.removeChild(privateFrame);
+
+      var iframe = document.createElement("iframe");
+      iframe.mozbrowser = "true";
+      iframe.onload = t.step_func(function() {
+        t.done();
```

Comment:
We should add an assert here that we have actually received the mozbrowsershowmodalprompt event.